[Known Issue][Obs & Sec] Upgrading to 9.3.x fails when a rule action contains oversized content (#6554)
elastic/docs-content: release-notes/elastic-observability/known-issues.md

% :::

> ::::{dropdown} Upgrading to 9.3.x fails when a rule action contains oversized content

Reviewer: double dropdown alert!
```suggestion
```

Reviewer: I didn't see a double ::::

Author: oh yeah, I just fixed that [here](https://github.com/elastic/docs-content/pull/6554#discussion_r3276724030). : )

> ::::{dropdown} Upgrading to 9.3.x fails when a rule action contains oversized content

Author: ```suggestion
```

::::{dropdown} Upgrading to 9.3.x fails when a rule action contains oversized content

Applies to: {{stack}} 9.3.0, 9.3.1, 9.3.2, 9.3.3, 9.3.4

**Details**

> Upgrading from 9.2.x to 9.3.x can fail if any Observability alerting rule has a connector action whose parameter values are larger than 32,766 bytes. Common examples include email message bodies or HTML templates, large webhook payloads, or Slack messages built from verbose templates.

Reviewer: ```suggestion
Upgrading from 9.2.x to 9.3.x can fail if a rule (including Observability alerting rules) has a connector action whose parameter values are larger than 32,766 bytes. Common examples include email message bodies or HTML templates, large webhook payloads, or Slack messages built from verbose templates.
```
Would make it more similar to the security version, and a bit more general

% :::

::::{dropdown} Upgrading to 9.3.x fails when a rule action contains oversized content
::::{dropdown} Upgrading to 9.3.x fails when a rule action contains oversized content

Applies to: {{stack}} 9.3.0, 9.3.1, 9.3.2, 9.3.3, 9.3.4

**Details**

Upgrading from 9.2.x to 9.3.x can fail if any Observability alerting rule has a connector action whose parameter values are larger than 32,766 bytes. Common examples include email message bodies or HTML templates, large webhook payloads, or Slack messages built from verbose templates.

During the upgrade, {{kib}} migrates rule saved objects to a new internal mapping. Any oversized action parameter value causes the migration to abort with an error similar to:

```
Flattened field [alert.actions.params] contains one immense field whose keyed encoding is longer than the allowed max length of 32766 bytes
```

**Workaround**

If the upgrade has failed with this error, identify rules that use connectors with large content (particularly email, webhook, and Slack connectors) and shorten the action parameter values, such as message bodies or HTML templates. Then retry the upgrade.

> To check rule action parameter sizes before upgrading, query the `.kibana_alerting_cases_*` index directly in {{es}} and inspect the `alert.actions.params` fields for any values that approach or exceed 32,766 bytes.

Author: Will be updating this suggestion with the info that comes from https://github.com/elastic/docs-content-internal/issues/1196#issuecomment-4482263369.

Author: Removing for now since don't know the actual query. Will update later once I have more info. 

```suggestion
```
